tests/org.obeonetwork.m2doc.tests/src/org/obeonetwork/m2doc/util/test/TemplateConfigUtilTest.java

```diff
@@ -0,0 +1,344 @@
+
+        CTProperty propV1 = props.getProperty("m:var:v1");
+        assertEquals("string", propV1.getLpwstr());
+
+        CTProperty propV2 = props.getProperty("m:var:v2");
+        assertEquals("ecore::EClassifier", propV2.getLpwstr());
+
+        CTProperty propV3 = props.getProperty("m:var:v3");
+        assertEquals("test::UnboundType", propV3.getLpwstr());
+    }
+
+    @Test
+    public void testIsValidVariableName() {
+        assertTrue(TemplateConfigUtil.isValidVariableName("v"));
+        assertTrue(TemplateConfigUtil.isValidVariableName("_valid"));
+        assertTrue(TemplateConfigUtil.isValidVariableName("valid_"));
+        assertTrue(TemplateConfigUtil.isValidVariableName("valid_123"));
+        assertTrue(TemplateConfigUtil.isValidVariableName("_VALID_1354"));
+
+        assertFalse(TemplateConfigUtil.isValidVariableName(null));
+        assertFalse(TemplateConfigUtil.isValidVariableName(""));
+        assertFalse(TemplateConfigUtil.isValidVariableName("not valid"));
+        assertFalse(TemplateConfigUtil.isValidVariableName("é"));
+        assertFalse(TemplateConfigUtil.isValidVariableName("3invalid"));
+        assertFalse(TemplateConfigUtil.isValidVariableName("-inv"));
+        assertFalse(TemplateConfigUtil.isValidVariableName("not-valid"));
+        assertFalse(TemplateConfigUtil.isValidVariableName("dfg$fsd"));
+    }
+
+    @Test
+    public void testIsValidTypeName() {
+        assertTrue(TemplateConfigUtil.isValidTypeName("string"));
+        assertTrue(TemplateConfigUtil.isValidTypeName("a::v"));
+        assertTrue(TemplateConfigUtil.isValidTypeName("_valid::Abc"));
+        assertTrue(TemplateConfigUtil.isValidTypeName("valid_::abc"));
+        assertTrue(TemplateConfigUtil.isValidTypeName("valid_123::abc"));
+        assertTrue(TemplateConfigUtil.isValidTypeName("_VALID_1354::_Type_Name_with_àccénts"));
+        assertTrue(TemplateConfigUtil.isValidTypeName("éàç_accented_chars_ok::_34"));
+
+        assertFalse(TemplateConfigUtil.isValidTypeName(null));
+        assertFalse(TemplateConfigUtil.isValidTypeName(""));
+        assertFalse(TemplateConfigUtil.isValidTypeName("int"));
+        assertFalse(TemplateConfigUtil.isValidTypeName("real"));
+        assertFalse(TemplateConfigUtil.isValidTypeName("boolean"));
+        assertFalse(TemplateConfigUtil.isValidTypeName("object"));
+        assertFalse(TemplateConfigUtil.isValidTypeName("date"));
+        assertFalse(TemplateConfigUtil.isValidTypeName("not valid::fds"));
+        assertFalse(TemplateConfigUtil.isValidTypeName("3invalid::fds"));
+        assertFalse(TemplateConfigUtil.isValidTypeName("-inv::fds"));
+        assertFalse(TemplateConfigUtil.isValidTypeName("not-valid::fds"));
+    }
+
+    @Before
+    public void setUp() {
+        EPackage.Registry.INSTANCE.put(EcorePackage.eNS_URI, EcorePackage.eINSTANCE);
+    }
+
+    @After
+    public void tearDown() {
+        EPackage.Registry.INSTANCE.remove(EcorePackage.eNS_URI);
```

Comment:
I don't want to be the test which gets executed after this one and which might get the Ecore EPackage un-registered ;)

I'm curious why you need to register/unregister EcorePackage at all.